### PR TITLE
Fix release script

### PR DESCRIPTION
**Pull request by say4n**
Fix release script to use CLI arguments instead of the hardcoded ones.

### CODE REVIEW

**Comment by say4n** [comment]
I will update `HISTORY.md` once #214 is merged.

**Comment by say4n** [comment]
Alright, now I will rebase on top of the current master and then update HISTORY.md accordingly.

File: scripts/ensmallen-release.sh [hunk 1]
```diff
@@ -132,7 +133,7 @@ git add HISTORY.md;
 git commit -m "Update and release version $MAJOR.$MINOR.$PATCH.";
 
 changelog_str=`cat HISTORY.md |\
-    awk '/^### /{f=0} /^### ensmallen 2.13.0: "Automatically Automated Automation"/{f=1} f{print}' |\
+    awk '/^### /{f=0} /^### ensmallen $MAJOR.$MINOR.$PATCH: "$version_name"/{f=1} f{print}' |\
```
Comment (rcurtin) [inline scripts/ensmallen-release.sh:136]:
```suggestion
    awk '/^### /{f=0} /^### ensmallen '"$MAJOR"'.'"$MINOR"'.'"$PATCH"': '"$version_name"'/{f=1} f{print}' |\
```
I believe that we'll have to do this in order to get the `MAJOR`, `MINOR`, and `PATCH` variables to substitute into the awk program correctly.  Correct me if I'm wrong on that.

Comment (say4n) [reply scripts/ensmallen-release.sh:136]: Ah my bad, I didn't test the script! 😛

Comment (rcurtin) [reply scripts/ensmallen-release.sh]: No worries!  If you want, you could test by running it, then just closing the PR that it opens. :+1:

Comment (say4n) [reply scripts/ensmallen-release.sh]: Ok sure! :)

Comment (say4n) [reply scripts/ensmallen-release.sh]: One thing of note though, when I created the PR, I had to be on the master branch of the upstream repo (cloned from mlpack, not my fork).

Comment (rcurtin) [reply scripts/ensmallen-release.sh]: Oh, right.  Well, in that case, we can try to merge this and I suppose we can test it then. :)

Comment (say4n) [reply scripts/ensmallen-release.sh]: Haha, okay! :)

Comment (rcurtin) [reply scripts/ensmallen-release.sh]: Yeah, we can always just close the test PR, and if it's wrong, we can make another attempt at fixing the script.  I realize I wrote it in such a way that it needs to be on the master branch, and that makes it a bit hard to debug. :)

File: scripts/ensmallen-release.sh [hunk 2]
```diff
@@ -23,7 +23,8 @@ if [ "$#" -gt 5 ]; then
 fi
 
 # Make sure that the branch is clean.
-lines=`git diff | wc -l`;
+# Truncate leading whitespaces since wc -l on MacOS adds an extra \t.
```
Comment (zoq) [inline scripts/ensmallen-release.sh:26]: I think it's macOS but don't think people will be confused.

Comment (coatless) [reply scripts/ensmallen-release.sh:26]: It's `macOS`... No one will be confused. just don't name it OS X 😉

Comment (rcurtin) [reply scripts/ensmallen-release.sh:26]: Is it a roman numeral or not?  Do I pronounce it X or 10?  Question for the ages :smile:

**Comment by rcurtin** [comment]
Thanks @say4n!

**Pull request merged**